saleor/dashboard/product/forms.py

```diff
@@ -137,7 +137,53 @@ def clean(self):
         return data
 
 
-class ProductForm(forms.ModelForm):
+class AttributesMixin(object):
+    """Form mixin that dynamically adds attribute fields."""
+
+    available_attributes = ProductAttribute.objects.none()
+
+    # Name of a field in self.instance that hold attributes HStore
+    model_attributes_field = None
+
+    def __init__(self, *args, **kwargs):
+        if not self.model_attributes_field:
+            raise Exception(
+                'model_attributes_field must be set in subclasses of '
+                'AttributesMixin.')
+
+    def prepare_fields_for_attributes(self):
+        initial_attrs = getattr(self.instance, self.model_attributes_field)
+        for attribute in self.available_attributes:
+            field_defaults = {
+                'label': attribute.name, 'required': False,
+                'initial': initial_attrs.get(str(attribute.pk))}
+            if attribute.has_values():
+                field = ModelChoiceOrCreationField(
+                    queryset=attribute.values.all(), **field_defaults)
+            else:
+                field = forms.CharField(**field_defaults)
+            self.fields[attribute.get_formfield_name()] = field
+
+    def iter_attribute_fields(self):
+        for attr in self.available_attributes:
+            yield self[attr.get_formfield_name()]
```

Comment:
Shouldn't it be rather `self[attr.get_formfield_name()]`?

Reply:
You can access fields in this way as well. It was here before so I didn't change it.